### ISSUE TITLE
Optionally use 'Yive's Mirror' for Spigot / Vanilla / Bukkit / PaperSpigot 

### ISSUE THREAD

**Issue by mhsjlw** [voxpupuli/puppet-minecraft#24]
A possible idea would be to transition the current CraftBukkit installation to options for Spigot, Vanilla, Bukkit or PaperSpigot with corresponding version numbers on 'Yive's Mirror' which makes daily bleeding builds of each server softwares.

The direct link is https://tcpr.ca/ but is not officially supported by any of the above server softwares. 

It may not be worth the work or it may not be a feature that people currently want, but for the future it may be easier than having a user use direct links.

**Comment by timatooth** [comment]
@mhsjlw That would be nice, I've started playing around with this module and was wondering the same for supporting different server types such as Forge, Sponge, Spigot etc. The problem is with the ephemeral nature of mirrors which people setup especially given the ancient history of Bukkit.

```puppet
class { 'minecraft':
  source       => 'http://tcpr.ca/files/spigot/spigot-1.10.2-R0.1-SNAPSHOT-latest.jar',
  heap_size  => 1024,
  difficulty    => 2,
  motd          => 'A minecraft server... Again.',
  ops             => [ 'Timatooth', ]
}
```
to something like:

```puppet
class { 'minecraft':
  type           => 'spigot', #can use this attribute for server specific files e.g spigot.yaml
  version      => 'spigot-1.10.2-R0.1-SNAPSHOT',
  heap_size  => 1024,
  ...
}
The skeptic me says these mirrors come and go so often. It may not be worthwhile writing logic around them for convenience only to find they later break.

**Comment by mhsjlw** [comment]
Absolutely. I really your idea of setting a type and the specific version, but in production it doesn't really work. The example you gave would still require the user to enter a hard coded path because not all mirrors store files in the same place. Another option could be going straight to the source and downloading jars of off the mods CI although that might be a little unstable. Let me know what you think
